Rdkafka.xs

```diff
@@ -288,6 +288,74 @@ krd_flush(rdk, timeout_ms)
     CODE:
         rd_kafka_flush(rdk->rk, timeout_ms);
 
+int
+krd_producev(rdk, params)
+        rdkafka_t* rdk
+        HV* params
+    PREINIT:
+        rd_kafka_headers_t *hdrs;
+        SV **tsv, **psv, **msv, **ksv, **vsv, **hsv;
+        int msgflags, partition;
+        char *value, *topic, *key;
+        STRLEN vlen, klen;
+    CODE:
+        tsv = hv_fetchs(params, "topic", 0);
+        if (tsv == NULL || !SvOK(*tsv))
+            croak("topic must be defined");
+        topic = SvPVbyte_nolen(*tsv);
+
+        psv = hv_fetchs(params, "partition", 0);
+        if (psv == NULL)
+            partition = RD_KAFKA_PARTITION_UA;
+        else if (SvIOK(*psv))
+            partition = SvIV(*psv);
+        else
+            croak("partition must be a number");
+
+        msv = hv_fetchs(params, "msgflags", 0);
+        if (msv == NULL)
+            msgflags = 0;
+        else if (SvIOK(*msv))
+            msgflags = SvIV(*msv);
+        else
+            croak("msgflags must be a number");
+
+        ksv = hv_fetchs(params, "key", 0);
+        if (ksv != NULL && SvOK(*ksv)) {
+            key = SvPVbyte(*ksv, klen);
+        }
+        else {
+            key = NULL; klen = 0;
+        }
+
+        vsv = hv_fetchs(params, "value", 0);
+        if (vsv != NULL && SvOK(*vsv)) {
+            value = SvPVbyte(*vsv, vlen);
+        }
+        else {
+            value = NULL; vlen = 0;
+        }
+
+        hdrs = rd_kafka_headers_new(0);
```

Comment:
how do we handle malloc failures here?

Reply:
As far as I can tell, the library has its own [`malloc`](https://github.com/confluentinc/librdkafka/blob/master/src/rd.h#L139) that calls `assert` from the C library on failure.

Reply:
yeah, just noticed